test/classes/delete-free/lifetimes/bad-return-borrow-owned.chpl

```diff
@@ -0,0 +1,77 @@
+pragma "safe"
+module l10 {
+
+class MyClass {
+  var x:int;
+}
+record Rinit {
+  pragma "owned"
+  var c:MyClass;
+  proc init(c:MyClass) {
+    this.c = c;
```

Comment:
Hm how is this supposed to work? My understanding is that 'c' is not supposed to escape because it is not "return scope"; however it does because it is stashed in 'this.c' and 'this' escapes. ???

Reply:
yes it magically turned into an "owned" thing; once we have owned/borrowed/raw types this example can be more reasonable & the checker can be more complete.